File: backend/src/main/java/com/ztpai/backend/profession/Profession.java
package com.ztpai.backend.profession;

import com.ztpai.backend.ability.Ability;
import com.ztpai.backend.attribute.Attribute;
import com.ztpai.backend.socialClass.SocialClass;
import com.ztpai.backend.talent.Talent;
import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

@Getter
@Setter(AccessLevel.PRIVATE)
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "profession")
public class Profession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(
            name = "name",
            nullable = false,
            unique = true
    )
    private String name;

    private String path;

    private String description;

    private String socialStatus;

    private String equipment;

    private int level;

    @OneToOne
    private SocialClass socialClass;

    @OneToMany
    private Set<Ability> abilitySet;

    @OneToMany
    private Set<Talent> talentSet;

    @OneToMany
    private Set<Attribute> attributeSet;
}

File: frontend/src/app/_models/profession.ts
import {SocialClass} from "./socialClass";
import {Ability} from "./ability";
import {Attribute} from "./attribute";
import {Talent} from "./talent";

export class Profession{
  name!: string;
  path!: string;
  description! : string;
  socialStatus!: string;
  equipment!: string;
  level! : number;
  socialClass!: SocialClass;
  abilitySet!: Ability[];
  talentSet!: Talent[];
  attributeSet!: Attribute[];
}
